serverlessworkflow/modules/ROOT/pages/data-index/data-index-service.adoc

```diff
@@ -0,0 +1,241 @@
+
+[#data-index-service]
+== {data_index_ref} service deployment.
+
+{data_index_ref} service can be deployed referencing directly a distributed {data_index_ref} image. There are different images provided that take into account what persistence layer is required in each case.
+In each distribution, there are some properties to configure things like the connection with the database or the communication with other services.
+The goal is to configure the container to allow to process ProcessInstances and Jobs *events* that incorporate the related audit data, to index and store that in the database and finally, to provide the xref:data-index/data-index-core-concepts.adoc#data-index-graphql[{data_index_ref} GraphQL] endpoint to consume it.
+
+[#data-index-service-distributions]
+=== {data_index_ref} distributions
+{data_index_ref} service can be deployed referencing directly a distributed {data_index_ref} image. Here there are the different {data_index_ref} image distributions that can be found in link:{distributions_url}[Quay.io/kiegroup]:
+
+image::data-index/data-index-distributions.png[Image of data-index different available distributions]
+
+=== Data Index Service deployment
+
+When the Data Index Service is deployed, a `configMap` resource is created for the `application.properties` configuration of the Data Index Service.
+
+The name of the `configMap` resource consists of the name of the Data Index Service and the suffix `-properties`, as shown in the following example:
+
+.Example `configMap` resource generated during Data Index Service deployment
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: data-index-properties
+data:
+  application.properties : |-
+    property1=value1
+    property2=value2
+----
+
+The `application.properties` data of the `configMap` resource is mounted in a volume to the container of the Data Index Service. Any runtime properties that you add to the `application.properties` section override the default application configuration properties of the Data Index Service.
+
+When the `application.properties` data of the `configMap` is changed, a rolling update modifies the deployment and configuration of the Data Index Service.
+
+---
+
+=== PostgreSQL persistence with the Data Index Service
+
+Data Index Service does not initialize its database schema automatically. To initialize the database schema, you need to either enable automatic schema initialization using Hibernate or enabling flyway migration enabling QUARKUS_FLYWAY_MIGRATE_AT_START.
+
+Automatic schema initialization using Hibernate can be enabled using the `quarkus.hibernate-orm.database.generation` property. For possible values of `quarkus.hibernate-orm.database.generation` property, see https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.database.generation[Quarkus docs for database generation options]. The value can also be set as an environment variable `QUARKUS_HIBERNATE_ORM_DATABASE_GENERATION`.
```

Comment:
hm, I think we dont need to mention this, might be confusing to end users, lets mention only Flyway